plugins/modules/cloud/scaleway/scaleway_container_registry.py

```diff
@@ -0,0 +1,267 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Scaleway Container registry management module
+#
+# Copyright (c) 2022, Guillaume MARTINEZ <lunik@tiwabbit.fr>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: scaleway_container_registry
+short_description: Scaleway Container registry management module
+version_added: 5.8.0
+author: Guillaume MARTINEZ (@Lunik)
+description:
+  - This module manages container registries on Scaleway account.
+extends_documentation_fragment:
+  - community.general.scaleway
+  - community.general.scaleway_waitable_resource
+
+
+options:
+  state:
+    type: str
+    description:
+      - Indicate desired state of the container regitry.
+    default: present
+    choices:
+      - present
+      - absent
+
+  project_id:
+    type: str
+    description:
+      - Project identifier.
+    required: true
+
+  region:
+    type: str
+    description:
+      - Scaleway region to use (for example C(fr-par)).
+    required: true
+    choices:
+      - fr-par
+      - nl-ams
+      - pl-waw
+
+  name:
+    type: str
+    description:
+      - Name of the container registry.
+    required: true
+
+  description:
+    description:
+      - Description of the container registry.
+    type: str
+
+  privacy_policy:
+    type: str
+    description:
+      - Default visibility policy.
+      - Everyone will be able to pull images from a C(public) registry.
+    choices:
+      - public
+      - private
+    default: private
+'''
+
+EXAMPLES = '''
+- name: Create a container registry
+  community.general.scaleway_container_registry:
+    project_id: '{{ scw_project }}'
+    state: present
+    region: fr-par
+    name: my-awesome-container-registry
+  register: container_registry_creation_task
+
+- name: Make sure container regitry is deleted
+  community.general.scaleway_container_registry:
+    project_id: '{{ scw_project }}'
+    state: absent
+    region: fr-par
+    name: my-awesome-container-registry
+'''
+
+RETURN = '''
+container_registry:
+    description: The container registry informations.
+    returned: when I(state=present)
+    type: dict
+    sample: {
+      "created_at": "2022-10-14T09:51:07.949716Z",
+      "description": "Managed by Ansible",
+      "endpoint": "rg.fr-par.scw.cloud/my-awesome-registry",
+      "id": "0d7d5270-7864-49c2-920b-9fd6731f3589",
+      "image_count": 0,
+      "is_public": false,
+      "name": "my-awesome-registry",
+      "organization_id": "10697b59-5c34-4d24-8d15-9ff2d3b89f58",
+      "project_id": "3da4f0b2-06be-4773-8ec4-5dfa435381be",
+      "region": "fr-par",
+      "size": 0,
+      "status": "ready",
+      "status_message": "",
+      "updated_at": "2022-10-14T09:51:07.949716Z"
+    }
```

Comment:
This works, of course, but for the sake of consistency/readability, it might be interesting to express it as YAML-ish instead of JSON-ish.

Reply:
Fixed in ac03fd5c5b30d99f508cb2699f9ff3513fabc44a